lib/stm32/fdcan.c

```diff
@@ -0,0 +1,891 @@
+ * @param nr number of filter to be configured
+ * @param id_list_mode mode in which id1 and id2 are used to match the rule, see 
+ * FDCAN_SFT_<mode> defines.
+ * @param id1 standard ID for matching. Used as exact value, lower bound or bit 
+ * pattern depending on matching mode selected
+ * @param id2 standard ID or bitmask. Used as exact value, upper bound or bit mask 
+ * depending on matching mode selected
+ * @param action Action performed if filtering rule matches frame ID, see 
+ * FDCAN_SFEC_<action> defines.
+ * @returns FDCAN_E_OK if filter was set up properly, 
+ * FDCAN_E_INVALID if FDCAN block base address is unknown and 
+ * FDCAN_E_OUTOFRANGE is filter number is larger than previosly configured.
+ */
+int fdcan_std_filter_set(uint32_t canport, uint32_t nr, 
+             uint8_t id_list_mode, uint32_t id1, uint32_t id2,
+		     uint8_t action)
+{
+	if ((FDCAN_CCCR(canport) & (FDCAN_CCCR_INIT | FDCAN_CCCR_CCE)) != 
+			(FDCAN_CCCR_INIT | FDCAN_CCCR_CCE))
+	{
+
+	}
+	struct FDCAN_MessageRAM * ram = fdcan_get_msgram_addr(canport);
+
+	if (!ram)
+	{
+		return FDCAN_E_INVALID;
+	}
+
+	uint32_t rxgfc_lss_current = ((FDCAN_RXGFC(canport) & FDCAN_RXGFC_LSS_MASK) 
+			>> FDCAN_RXGFC_LSS_SHIFT);
+
+	if (nr >= rxgfc_lss_current)
+	{
+		return FDCAN_E_OUTOFRANGE;
+	}
+
+	ram->LFSSA[nr].type_id1_conf_id2 = 
+		((id_list_mode << FDCAN_SFT_SHIFT) & FDCAN_SFT_MASK)
+		| (action & FDCAN_SFEC_MASK)
+		| ((id1 << FDCAN_SFID1_SHIFT) & FDCAN_SFID1_SHIFT)
+		| ((id2 << FDCAN_SFID2_SHIFT) & FDCAN_SFID2_SHIFT);
+
+	return 0;
+}
+
+/** Configure filter rule for extended ID frames.
+ * Sets up filter rule for frames having extended ID. Each FDCAN block can
+ * have its own set of filtering rules. It is only possible to configure as
+ * many filters as was configured previously using @see fdcan_filter_init.
+ * @param canport FDCAN block base address
+ * @param nr number of filter to be configured
+ * @param id_list_mode mode in which id1 and id2 are used to match the rule, see 
+ * FDCAN_EFT_<mode> defines.
+ * @param id1 extended ID for matching. Used as exact value, lower bound or bit 
+ * pattern depending on matching mode selected
+ * @param id2 extended ID or bitmask. Used as exact value, upper bound or bit mask 
+ * depending on matching mode selected
+ * @param action Action performed if filtering rule matches frame ID, see 
+ * FDCAN_EFEC_<action> defines.
+ * @returns FDCAN_E_OK if filter was set up properly, 
+ * FDCAN_E_INVALID if FDCAN block base address is unknown and 
+ * FDCAN_E_OUTOFRANGE is filter number is larger than previosly configured.
+ */
+int fdcan_ext_filter_set(uint32_t canport, uint32_t nr, 
+             uint8_t id_list_mode, uint32_t id1, uint32_t id2,
+		     uint8_t action)
+{
+	struct FDCAN_MessageRAM * ram = fdcan_get_msgram_addr(canport);
+
+	if (!ram)
+	{
+		return FDCAN_E_INVALID;
+	}
+
+	uint32_t rxgfc_lse_current = ((FDCAN_RXGFC(canport) & FDCAN_RXGFC_LSE_MASK) 
+			>> FDCAN_RXGFC_LSE_SHIFT);
```

Comment:
this is just reading back what the user set themselves earlier right?  this is nice big secure safe defensive never trust anyone code, but it's very much not what we've done so far with "do what the users asked to do, with a neat api and as little fuss as possible"  I would consider whether this is really necessary or beneficial.